### PR TITLE
[5.3] Check for api_token in headers.

### PR DESCRIPTION
**Pull request by srmklive**

### CODE REVIEW

**Comment by srmklive** [comment]
@GrahamCampbell Can you advise on how to write tests in this regard? I have written some locally, but they keep failing if a pass an api_key in headers.

**Comment by crynobone** [comment]
You should be using the `Authorization

```
Authorization: Bearer {token}
```

File: src/Illuminate/Auth/TokenGuard.php [hunk 1]
```diff
@@ -32,17 +32,24 @@ class TokenGuard implements Guard
...
+     * @var string
+     */
+    protected $headerKey;
+
+    /**
      * Create a new authentication guard.
      *
      * @param  \Illuminate\Contracts\Auth\UserProvider  $provider
      * @param  \Illuminate\Http\Request  $request
-     * @return void
```
Comment (GrahamCampbell) [inline src/Illuminate/Auth/TokenGuard.php]: Please revert this bad change.

**Comment by srmklive** [comment]
@crynobone there is already a test specifying the parameters you mentioned.

**Comment by crynobone** [comment]
> there is already a test specifying the parameters you mentioned.

@srmklive so why do you need to add a non standard HTTP Request Header when you can be using `Authorization` header?

**Comment by taylorotwell** [comment]
Use the Authorization header which is intended for this purpose.

**Pull request closed without merging**